### ISSUE TITLE
Why tempo trace queries are missing after 30min?

### ISSUE THREAD

**Issue by npuichigo** [grafana/tempo#3155]
I use docker-compose to run tempo with grafana, and it works well to show the trace span.
```yaml
  tempo:
    image: grafana/tempo:2.1.1
    command:
      - "-target=all"
      - "-storage.trace.backend=local"
      - "-storage.trace.local.path=/var/tempo"
      - "-ingester.complete-block-timeout=240h"
      - "-ingester.max-block-duration=240h"
      - "-auth.enabled=false"
    volumes:
      - tempo-data:/var/tempo
    ports:
      - "14250:14250"
    depends_on:
      - loki
    logging: *default-logging
    restart: on-failure

volumes:
  tempo-data:
    driver: local
```
However, after about 30mins, I cannot find previous trace spans before 30mins ago.
![image](https://github.com/grafana/tempo/assets/11533479/9a79799d-8b28-4502-8b74-e7adc2e159b9)

I tried to set ingester parameters but failed. Any idea about this?

**Comment by npuichigo** [comment]
Updates: query last two days could find something really old, that's strange
![image](https://github.com/grafana/tempo/assets/11533479/e29cf827-997d-4363-aa98-6a0bbc1e9d94)

**Comment by joe-elliott** [comment]
To control your retention set:

```
compactor:
  compaction:
    block_retention: <some go duration>
```

**Comment by npuichigo** [comment]
I checked the default for block-retention is 336 hours. It seems the issue is related to https://github.com/grafana/tempo/issues/2565 and has been addressed in 2.2
```
-compactor.compaction.block-retention duration
     Duration to keep blocks/traces. (default 336h0m0s)
```

**Comment by joe-elliott** [comment]
Ah, I see :+1: 

Are we good to close this?
